environments/magento2.base.yml

```diff
@@ -136,6 +136,39 @@ services:
       - warden
       - default
 
+  crontab:
```

Comment:
Would it work to add this as a Global Service rather than have a crontab per environment? If I'm understanding things correctly, this is looking at labels on running containers and essentially `exec`'ing into them to run what's been defined in the labels.

Reply:
I considered that but not without heavy modifications to the code (which I'm not comfortable doing).

The reason being is that if it was a global service, it would run any cron that had been started at one point after the image.  A `warden env down` would not work, as the global crontab would restart the cronjob of that env.

Reply:
That makes sense. I think `down` would be ok (it destroys the container) but `stop` would not since they are still there. I spent a little time reading through the code for it myself late on Friday and thought it looked like it was listening to stop/die events to de-register, but when I ran a little test, it panned out as you described, so I'm not 100% what the event logic was for (possibly something more directly tied to Rancher behaviour). Since it's either starting/restarting/stopping to execute a job, it makes sense it'd be difficult to alter the code in this way.

Found this little gem which is similar: https://github.com/mcuadros/ofelia but it doesn't currently watch docker events (only reads what's there when it starts up) so it wouldn't work in this way either (i.e. one orchestration container looking at labels across all containers).

Going to try and run another test this week and see what happens if I have Warden remove the cron job container in an environment entirely when running `stop` and see what happens. That might work to allow using this, but with a single orchestration container.

Reply:
It sounds like my being nerd sniped has nerd sniped you